### PR TITLE
Add TypeScript typings

### PR DESCRIPTION
**Pull request by yassernasc**

### CODE REVIEW

**Review by kasperisager — CHANGES_REQUESTED** [review]
Let's declare the public instance properties as well, such as `Version#major`, `Version#minor`, etc.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    state?: { position: number; partial: boolean }
+  ): Version
+
+  export function compare(a: Version, b: Version): number
+}
+
+export class Comparator {
+  constructor(operator: number, version: Version)
+
+  operator: string
```
Comment (kasperisager) [inline index.d.ts]:
```suggestion
  operator: number
```

**Review by kasperisager — CHANGES_REQUESTED** [review]

**Review by kasperisager — APPROVED** [review]

**Comment by yassernasc** [comment]
thanks for you patience @kasperisager, that one mistake was really silly 😅

**Comment by kasperisager** [comment]
No worries! I could have easily made the same typo myself 😄

**Pull request merged**